bin/DESeq2.R

```diff
@@ -173,19 +190,79 @@ if (!is.null(opt$relevel)) {
 }
 
 # Run DESeq function
-cds <- DESeqDataSetFromMatrix( countData =count.table, colData =metadata, design = eval(parse(text=as.character(design[[1]]))))
-cds <- DESeq(cds,  parallel = FALSE)
+if (opt$input_type == "rawcounts") {
+    cds <- DESeqDataSetFromMatrix( countData =count.table, colData =metadata, design = eval(parse(text=as.character(design[[1]]))))
+    cds <- DESeq(cds,  parallel = FALSE)
+} else if (opt$input_type %in% c("rsem", "salmon")) {
+    ## Create a dataframe which consists of both the gene id and the transcript name
+    gtf <- rtracklayer::import(opt$gtf)
+    gtf <- as.data.frame(gtf, header=T)
+    # TODO: For some gtf files, transcript_id does not work!!
+    tx2gene_gtf <- gtf[c("transcript_id", "gene_id")]
+    tx2gene_gtf <- distinct(tx2gene_gtf)
+    tx2gene_gtf[] <- lapply(tx2gene_gtf, function(x) gsub("\\.\\d+", "", x))
+    colnames(tx2gene_gtf) <- c("transcript_id", "gene_id") #, "TXID"
+    gene_names <- gtf[c("gene_id", "gene_name")]
+    colnames(gene_names) <- c("Ensembl_ID", "gene_name")
+    gene_names <- distinct(gene_names)
+    rownames(gene_names) <- gene_names[,1]
+
+    if (opt$input_type == "rsem") {
+        files <- file.path(gsub("/$", "", path_count_table), paste0(qbicCodes, ".genes.results"))
```

Comment:
Ok, so here there is not one merged gene count table, but a folder where the count tables for the individual samples are contained, right?

Reply:
Yes, both rsem and salmon produce one file for each sample (salmon saves the files nested in a subfolder per sample). Will comment in the code